public/pages/Findings/components/FindingDetailsFlyout.tsx

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { Component } from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+import {
+  EuiBadge,
+  EuiBadgeGroup,
+  EuiButton,
+  EuiCodeBlock,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiEmptyPrompt,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiFormRow,
+  EuiInMemoryTable,
+  EuiLink,
+  EuiOverlayMask,
+  EuiPopover,
+  EuiSpacer,
+  EuiTableFieldDataColumnType,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import { OpenSearchService } from '../../../services';
+import { renderTime } from '../../../utils/helpers';
+import { DEFAULT_EMPTY_DATA } from '../../../utils/constants';
+import { parseSeverityToOption } from '../../CreateDetector/components/ConfigureAlerts/utils/helpers';
+
+interface FindingDetailsFlyoutProps extends RouteComponentProps {
+  finding: Finding;
+  opensearchService: OpenSearchService;
+  closeFlyout: () => void;
+}
+
+interface FindingDetailsFlyoutState {}
+
+export default class FindingDetailsFlyout extends Component<
+  FindingDetailsFlyoutProps,
+  FindingDetailsFlyoutState
+> {
+  constructor(props: FindingDetailsFlyoutProps) {
+    super(props);
+    this.state = {
+      documents: [],
+      loading: false,
+    };
+  }
+
+  getDetector = async () => {
+    this.setState({ loading: true });
+    // TODO: Call the get detector API. Find the rule associated with the
+    //  finding to retrieve the rule details for display.
+    this.setState({ loading: false });
+  };
+
+  getDocument = async () => {
+    this.setState({ loading: true });
+    const {
+      finding: { index, related_doc_ids = [] },
+      opensearchService,
+    } = this.props;
+    try {
+      const response = await opensearchService.documentIdsQuery(index, related_doc_ids);
+      if (response.ok) {
+        this.setState({ documents: response.response.hits.hits || [] });
+      } else {
+        console.error('Failed to retrieve documents: ', response);
+      }
+    } catch (e) {
+      console.error('Failed to retrieve documents: ', e);
+    }
+    this.setState({ loading: false });
+  };
+
```

Comment:
Maybe not an issue for right now, but dashboards does provide a 'toasts' functionality for messages to users (errors, success).

Reply:
Yes I will use the toasts in later PRs once I have all the backend APIs integrated